### PR TITLE
Added update protocol action

### PR DESCRIPTION
**Pull request by thehenrytsai**

### CODE REVIEW

File: src/core/protocol-authorization.ts [hunk 1]
```diff
@@ -536,14 +536,19 @@ export class ProtocolAuthorization {
...
-        return [ProtocolAction.CoUpdate];
+        // else not initial write
+
+        if (await incomingRecordsWrite.isAuthoredByInitialRecordAuthor(tenant, messageStore)) {
+          // 'write', 'update' or 'co-update' action authorizes the incoming message
+          return [ProtocolAction.Write, ProtocolAction.CoUpdate, ProtocolAction.Update];
+        } else {
+          // Actors other than the initial record author must be authorized to 'co-update' the message
+          return [ProtocolAction.CoUpdate];
+        }
```
Comment (thehenrytsai) [inline src/core/protocol-authorization.ts:551]: No functional change, just refactor for readability.

**Review by csuwildcat — DISMISSED** [review]

File: tests/features/protocol-create-action.spec.ts [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+
+      const protocolsConfigureReply = await dwn.processMessage(alice.did, protocolsConfig.message);
+      expect(protocolsConfigureReply.status.code).to.equal(202);
+
+      // Verify Bob cannot create a record without a matching create rule
+      const bobFooBytes = TestDataGenerator.randomBytes(100);
+      const bobUnauthorizedWrite = await RecordsWrite.create(
+        {
+          signer       : Jws.createSigner(bob),
+          protocol     : protocolDefinition.protocol,
```
Comment (LiranCohen) [inline tests/features/protocol-create-action.spec.ts:133]: super nit: Do you mind adding carol as the recipient on this one as well so it more closely matches the successful message?

**Review by LiranCohen — APPROVED** [review]
🥇

File: tests/features/protocol-create-action.spec.ts (hunk 2 above)
Comment (thehenrytsai) [reply tests/features/protocol-create-action.spec.ts:134]: Sure thing, done!

**Review by LiranCohen — APPROVED** [review]

**Pull request merged**